open-api/lang-tools/src/test/groovy/io/deephaven/lang/completion/ChunkerGroovyTest.groovy

```diff
@@ -111,4 +112,10 @@ l.add(new SomeClass().<? super String>method(new Thing1<? super T>.Thing2<? supe
         // oddly parser handles classes and whatnot at this time.
 
     }
+    @Override
+    VariableProvider getVariables() {
+        return Mock(VariableProvider) {
+            _ * getVariableNames() >> []
+        }
```

Comment:
I'm unfamiliar w/ this syntax - what does this mean?

Reply:
it's spock mocks. `# * methodName(args?) >> valueToReturn`

The # can be any integer or _ to say "any amount", or a range, like 0..3
The method name is from the class passed to Mock(), the arguments (not here) can also be `_` to express "anything".

A return value is optional, and only needed if you want it to return a non-default value (null, 0, "", etc).

https://semaphoreci.com/community/tutorials/stubbing-and-mocking-in-java-with-the-spock-testing-framework